rest/src/main/scala/tamer/rest/RestConfiguration.scala

```diff
@@ -46,9 +54,23 @@ final case class RestConfiguration[
 }
 
 object RestConfiguration {
+  def apply[
+      R,
+      K,
+      V,
+      S
+  ](queryBuilder: RestQueryBuilder[R, S])(
+      pageDecoder: String => RIO[R, DecodedPage[V, S]]
+  )(transitions: State[K, V, S])(implicit k: Codec[K], v: Codec[V], s1: Codec[S], s2: HashableState[S], d: DummyImplicit) =
```

Comment:
why the `DummyImplicit`?

Reply:
ah! right! ok, followup question then: do we need this to be a case class? If not we could arrange the parameter lists in the class itself to help type inference

Reply:
Ah yeah good thinking I'll convert this into a regular class